### DISCUSSION TITLE
How to do alignment with PyG format xyz?

### DISCUSSION THREAD

**Discussion by StefanIsSmart** [pyg-team/pytorch_geometric#9411] [Q&A]
When we are dealing with compounds, sometimes we need to predict the structure.
The question is how to align them with the batch.

For example, there are `pred_xyz` and `label_xyz`, we need to use SVD to align them and then use the MAEloss to calculate the loss of prediction.
In PyG, the format of `pred_xyz` and `label_xyz` with be : (N, 3).[In fact, N is the sum of nodes from G compounds (So the batch_size of compounds is G, and the batch_size of Node is N, **the length of every compound is different**)]

We cannot directly use SVD to align de `pred_xyz` and `label_xyz`, because that will lead to the `compound_(1)` will consider the coordinates of `compound_(2-G)` during the alignment.

I have to do that by for-loop, but it is very ineffective.

How could I do that with batch rather than for-loop?

**Comment by rusty1s** [comment]
The recommended way to do this is to convert your matrix into a 3-dimensional tensor with a batch dimension via `utils.to_dense_batch`. Note that you need to take care of proper padding treatment in this case.
